### PR TITLE
Breaking: revamp error

### PR DESCRIPTION
**Pull request by vilgotf**
* Fixes doctest thus removing `src/test/endpoints_test.rs`
* Moves `src/parsing/mod.rs` -> `src/parsing.rs` (rust 2018 idiom)
* Move `ParseError` to separate enum
* Renames `YoutubeResult` to `Result`, `YoutubeError` to `Youtube` (user should import as `use youtube_metadata::error::Error as YoutubeError`)

### CODE REVIEW

File: src/endpoints.rs [hunk 1]
```diff
@@ -22,10 +24,24 @@ use crate::types::VideoInformation;
...
-pub async fn get_video_information(url: &str) -> YoutubeResult<VideoInformation> {
+pub async fn get_video_information(url: &str) -> Result<VideoInformation> {
     let response = reqwest::get(url).await?;
     let response_text = response.text().await?;
 
     parse_video_information(&response_text)
 }
+
+#[cfg(test)]
+mod test {
```
Comment (Trivernis) [inline src/endpoints.rs]: I'd like to keep tests and implementation separated in different files

File: src/error.rs [hunk 2]
```diff
@@ -1,18 +1,63 @@
...
-#[derive(Debug, Error)]
-pub enum YoutubeError {
-    #[error(transparent)]
-    Reqwest(#[from] reqwest::Error),
+pub type Result<T> = std::result::Result<T, Youtube>;
 
-    #[error("Parse Error: {0}")]
-    ParseError(String),
+#[derive(Debug)]
+pub enum Youtube {
```
Comment (Trivernis) [inline src/error.rs]: I don't like the renaming. The purpose of the enum should be explicitly stated in the name. Calling it `Error` would be more fitting.

**Review by Trivernis — CHANGES_REQUESTED** [review]

**Comment by vilgotf** [comment]
Addressed changes.

Btw would adding a search function be in line with your vision of this crate? (if so I'd be willing to implement it)

**Comment by Trivernis** [comment]
@vilgotf yeah a search function would fit in this crate.

**Pull request merged**